frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.spec.ts

```diff
@@ -4,33 +4,714 @@ import { ChallengesettingsComponent } from './challengesettings.component';
+    expect(component.challenge.evaluation_details).toEqual(updatedEvaluationDetails);
+    expect(component.updateView).toHaveBeenCalled();
+    expect(component.challenge.evaluation_details).toEqual(updatedEvaluationDetails);
+    expect(globalService.showToast).toHaveBeenCalledWith('success', expectedSuccessMsg, 5);
+  });
+
+  it('should handle the API error for `editEvaluationCriteria` method', () => {
+    const updatedEvaluationDetails = 'Updated challenge evaluation details';
+    const expectedErrorMsg = {
+      error: 'Api error',
+    };
+    spyOn(apiService, 'patchUrl').and.returnValue(
+      new Observable((observer) => {
+        observer.error({ error: expectedErrorMsg.error });
+        observer.complete();
+        return { unsubscribe() {} };
+      })
+    );
+
+    component.editEvaluationCriteria();
+    expect(globalService.showModal).toHaveBeenCalled();
+    component.apiCall(updatedEvaluationDetails);
+    expect(apiService.patchUrl).toHaveBeenCalled();
+    expect(globalService.showToast).toHaveBeenCalledWith('error', expectedErrorMsg);
+  });
+
+  it('should show modal and successfully edit the terms and conditions', () => {
+    const updatedTermsAndConditions = 'Updated terms and conditions of challenge';
+    const expectedSuccessMsg = 'The terms and conditions are successfully updated!';
+    spyOn(apiService, 'patchUrl').and.returnValue(
+      new Observable((observer) => {
+        observer.next({ terms_and_conditions: updatedTermsAndConditions });
+        observer.complete();
+        return { unsubscribe() {} };
+      })
+    );
+
+    component.editTermsAndConditions();
+    expect(globalService.showModal).toHaveBeenCalled();
+    component.apiCall(updatedTermsAndConditions);
+    expect(apiService.patchUrl).toHaveBeenCalled();
+    expect(component.challenge.terms_and_conditions).toEqual(updatedTermsAndConditions);
+    expect(component.updateView).toHaveBeenCalled();
+    expect(component.challenge.terms_and_conditions).toEqual(updatedTermsAndConditions);
+    expect(globalService.showToast).toHaveBeenCalledWith('success', expectedSuccessMsg, 5);
+  });
+
+  it('should handle the API error for `editTermsAndConditions` method', () => {
+    const updatedTermsAndConditions = 'Updated terms and conditions of challenge';
+    const expectedErrorMsg = {
+      error: 'Api error',
+    };
+    spyOn(apiService, 'patchUrl').and.returnValue(
+      new Observable((observer) => {
+        observer.error({ error: expectedErrorMsg.error });
+        observer.complete();
+        return { unsubscribe() {} };
+      })
+    );
+
+    component.editTermsAndConditions();
+    expect(globalService.showModal).toHaveBeenCalled();
+    component.apiCall(updatedTermsAndConditions);
+    expect(apiService.patchUrl).toHaveBeenCalled();
+    expect(globalService.showToast).toHaveBeenCalledWith('error', expectedErrorMsg);
+  });
+
+  it('should show modal and successfully edit evaluation script', () => {
```

Comment:
Change to `should show modal and successfully update evaluation script`

Reply:
Done!